### PR TITLE
Make Container techn optional

### PR DESCRIPTION
**Pull request by netrounds-fredrik**
This change makes the `techn` argument to `Container` and friends optional, similar to how it is optional in `Rel`.

I am not very familiar with this project or plantuml in general, so please review carefully.

### CODE REVIEW

File: C4_Container.puml [hunk 1]
```diff
@@ -53,18 +53,18 @@ endlegend
...
+  !$container = ""
+  !if ($sprite != "")
+    !$container = $container + $getSprite($sprite)+'\n'
+  !endif
+  !$container = $container + '=='+$label
+  !if ($techn != "")
+    !$container = $container + '\n//<size:'+$TECHN_FONT_SIZE+'>['+$techn+']</size>//'
+  !endif
+  !if ($descr != "")
+    !$container = $container + '\n\n '+$descr
```
Comment (kirchsth) [inline C4_Container.puml:65]: space after `\n\n ` looks like an "old issue", can you please fix it and remove the space too?

File: C4_Container.puml [hunk 2]
```diff
@@ -53,18 +53,18 @@ endlegend
...
-!endif
-!if ($descr != "") && ($sprite == "")
-!return '=='+$label+'\n//<size:'+$TECHN_FONT_SIZE+'>['+$techn+']</size>//\n\n '+$descr
-!endif
-!if ($descr != "") && ($sprite != "")
-!return $getSprite($sprite)+'\n=='+$label+'\n//<size:'+$TECHN_FONT_SIZE+'>['+$techn+']</size>//\n\n '+$descr
-!endif
+  !$container = ""
+  !if ($sprite != "")
+    !$container = $container + $getSprite($sprite)+'\n'
```
Comment (kirchsth) [inline C4_Container.puml:58]: Can you add a space before and after the + operator that we have the same formatting in a line?

**Comment by kirchsth** [comment]
Hi @netrounds-fredrik

thank you for the PR.
Optional $techn is a good finding/idea (In the meantime C4Model has container and components with optional technology too)

Can you please update the Component and documentation too?

Related to documentation: 
I think it is enough if you delete the line 
" *techn .. technology argument is required, ..."
and replace the remaining (2) *techn with ?techn

Thank you and best regards
Helmut

**Review by kirchsth — CHANGES_REQUESTED** [review]

**Comment by netrounds-fredrik** [comment]
I have addressed the comments. Also made the changes to make the `techn` argument to `Component` optional to align the behavior.

**Review by kirchsth — APPROVED** [review]

**Comment by kirchsth** [comment]
@netrounds-fredrik thank you for the PR 
Helmut

**Pull request merged**